lib/system.nim

```diff
@@ -1871,6 +1871,21 @@ proc `$` *[Enum: enum](x: Enum): string {.magic: "EnumToStr", noSideEffect.}
   ## a ``$`` operator for a concrete enumeration is provided, this is
   ## used instead. (In other words: *Overwriting* is possible.)
 
+proc newString*[N](data: array[N, char]): string {.noSideEffect.} =
```

Comment:
This is just confusing. The type signature clearly suggests that the array is of a fixed size and not treated as null terminated. Use at least a name like ``newStringZ``.